### PR TITLE
feat(clerk-js,themes): Introduce internal polished theme and a clerk theme that enables opt out

### PR DESCRIPTION
**Pull request by desiprisg**
## Description

Have fancy css in a separate internal theme in order to keep the codebase cleaner, and enable opt-out via a dummy theme.

```
import { reset } from '@clerk/themes';

 <ClerkProvider
      appearance={{
        baseTheme: [reset, customTheme],
        variables: {
            ...
        },
      }}
      {...pageProps}
 >
 ```

This would also work as expected (only signIn would be reset):

```
import { reset } from '@clerk/themes';

 <ClerkProvider
      appearance={{
        signIn: [reset, customTheme],
        variables: {
            ...
        },
      }}
      {...pageProps}
 >
 ```

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

**Comment by desiprisg** [comment]
Transferring/adding more styles to `polishedAppearance` soon.

**Comment by desiprisg** [comment]
!preview

**Comment by clerk-cookie** [comment]
Hey @desiprisg, your preview is currently deploying and should be available shortly.

| Status | Preview | Updated (UTC) |
| :-- | :-- | :-- |
| :cook: _Deploying..._ | [Inspect](https://javascript-preview-7jufo87zx.clerkstage.dev) | Feb 07, 2024 05:52 PM |

File: packages/themes/src/themes/reset.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { experimental_createTheme } from '../createTheme';
+
+export const reset = experimental_createTheme({
+  //@ts-expect-error not public api
+  simpleStyles: true,
```
Comment (anagstef) [inline packages/themes/src/themes/reset.ts:5]: Can we rename this one to `resetStyles`?

Comment (anagstef) [reply packages/themes/src/themes/reset.ts:5]: Skip for now.

**Review by anagstef — APPROVED** [review]

**Pull request merged**